python/lsst/meas/extensions/trailedSources/NaivePlugin.py

```diff
@@ -159,6 +160,13 @@ def measure(self, measRecord, exposure):
 
         # Transform the second-moments to semi-major and minor axes
         Ixx, Iyy, Ixy = measRecord.getShape().getParameterVector()
```

Comment:
Can you clarify what you mean by this? In the errors I have looked at, both `measRecord.getShape()` and the moments are nans, so are you suggesting it would it be better if we check `measRecord.getShape()` first and use that to fail out instead of the moments?

Reply:
If `getShapeFlag()==True`, that tells you that the shape measurement algorithm had a problem and the results might not be trustworthy. It should certainly be set if the results are NaN, but it would also be set for other kinds of problems. In fact, since you have known cases of NaN moments, you should check whether the shape flag is set for those: I'll be it is. So, you can probably just use that, which will also guard against other bad inputs.

Reply:
Ah so then doing it like this instead before calculating the moment would be a good solution? 
``` 
        if measRecord.getShapeFlag():
            self.log.debug("Second moments are nan for measRecord: %s", measRecord.getId())
            self.flagHandler.setValue(measRecord, self.FAILURE.number, True)
            self.flagHandler.setValue(measRecord, self.SECOND_MOMENT.number, True)
            return
```

Also, looking at the fail function, if it gets an error it expects it to be `error.cpp`. Is that a specific property of errors that are raised in python?

Reply:
`error.cpp` looks like an old vestigial thing: it's part of the `lsst.pex.exceptions` errors that we've defined for C++/python exception compatibility. Please ask on #dm-science-pipelines about what it's supposed to be used for, because I can't find documentation on it: It looks like it's mostly used in older meas_base code and more recent probably cargo-culted code. 

You should just be able to call `fail(record, err)` and be done with it, where `err=MeasurementError("Second moments are ...", self.SECOND_MOMENT)`, I think.

Reply:
Note that flgaHandler docs are here: https://pipelines.lsst.io/v/daily/modules/lsst.meas.base/tasks_and_algorithms.html#using-a-flaghandler-with-python-plugins